src/scripts/eastern_kingdoms/eastern_plaguelands/naxxramas/boss_patchwerk.cpp

```diff
@@ -119,68 +119,55 @@ struct boss_patchwerkAI : public ScriptedAI
 
     void DoHatefulStrike()
     {
-        // The ability is used on highest HP target in melee
-        // current tank cannot be hit by hateful as long as there are other players in melee
-        
-        // todo: can it hit anything other than players?
-
-        Unit* mainTank = m_creature->GetVictim();
-        
-        // Shouldnt really be possible, but hey, weirder things have happened
-        if (!mainTank)
-            return;
-        ObjectGuid const& mainTankGuid = mainTank->GetObjectGuid();
+        // Hateful Strike: https://classic.wowhead.com/guides/patchwerk-naxxramas-raid-strategy
+        // 1) target players between second, third, and fourth on threat
+        // 2) only players within melee range
+        // 3) target player with the most health
 
-        Unit* pTarget = nullptr;
-        uint32 uiHighestHP = 0;
+        Unit* pTarget            = nullptr;
+        uint32 uiHighestHP       = 0;
         uint8 threatListPosition = 0;
 
         ThreatList const& tList = m_creature->GetThreatManager().getThreatList();
         for (const auto iter : tList)
         {
-            // Only top 4 players on threat in melee range are targetted.
-            if (threatListPosition > 3)
+            // Only second, third, and fourth form threat list can be targeted
+            if (threatListPosition > 4)
```

Comment:
Now it will target top 5, since we count from 0...

Reply:
Due to the sources I've linked, it should only target players between second, third, and fourth from threat list. So index 0 should be skipped. (There's a if which does that in below conditions).

Reply:
It does not sound plausible to me that it cant hit the main tank, otherwise the strategy would be to only send 1 tank and no melee dps, and the fight would be piss easy, literally tank and spank.

Reply:
https://old.reddit.com/r/classicwow/comments/jw3vlt/patchwerk_hateful_strike_clarification/

Reply:
Interresting. At https://classic.wowhead.com/guides/patchwerk-naxxramas-raid-strategy they claim main tank can't be targeted: _Understanding these three mechanics is key. It will never hit the person with the most threat, so that means the Main Tank is only taking damage from auto attacks._

Reply:
Just found video where all meeles are dead except the main tank https://youtu.be/pjUv7HfOPbE?t=563 but I can't tell if the tank only gets meele hits (maybe with crits) or also hateful strikes.

Reply:
After watching the video and doing some tests with vmangos, Patchwerk must be using hateful strike on the main tank, if no one else (from position 2-4 on thread list) is available in meele range. I've tested the fight on vmangos without the tank being targeted with hateful strike, and the damage on my test character never reached those high values like what the tank in the video had to cope with. The frequency of high damage occuring in the video seems to fit to a 1,5-2 seconds timer, like mentioned at wowhead.